diff --git a/node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js b/node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js
index e3b63d0..d485ff0 100644
--- a/node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js
+++ b/node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js
@@ -82,8 +82,12 @@ export default function ComponentCreator(path, hash) {
                     Object.keys(loadedModule)
                         .filter((k) => k !== 'default')
                         .forEach((nonDefaultKey) => {
-                        chunk[nonDefaultKey] =
-                            loadedModule[nonDefaultKey];
+                        try {
+                            chunk[nonDefaultKey] =
+                                loadedModule[nonDefaultKey];
+                        } catch (e) {
+                            // Skip non-writable properties (e.g. Function.length)
+                        }
                     });
                 }
                 // We now have this chunk prepared. Go down the key path and replace the
diff --git a/node_modules/@docusaurus/core/lib/commands/build.js b/node_modules/@docusaurus/core/lib/commands/build.js
index 2865e66..7c69768 100644
--- a/node_modules/@docusaurus/core/lib/commands/build.js
+++ b/node_modules/@docusaurus/core/lib/commands/build.js
@@ -113,7 +113,7 @@ async function buildLocale({ siteDir, locale, cliOptions, }) {
     const { outDir, plugins, siteConfig } = props;
     const router = siteConfig.future.experimental_router;
     // We can build the 2 configs in parallel
-    const [{ clientConfig, clientManifestPath }, { serverConfig, serverBundlePath }] = await utils_3.PerfLogger.async('Creating webpack configs', () => Promise.all([
+    const [{ clientConfig, additionalPluginConfigs, clientManifestPath }, { serverConfig, serverBundlePath }] = await utils_3.PerfLogger.async('Creating webpack configs', () => Promise.all([
         getBuildClientConfig({
             props,
             cliOptions,
@@ -125,10 +125,10 @@ async function buildLocale({ siteDir, locale, cliOptions, }) {
     // Run webpack to build JS bundle (client) and static html files (server).
     await utils_3.PerfLogger.async('Bundling with Webpack', () => {
         if (router === 'hash') {
-            return (0, utils_2.compile)([clientConfig]);
+          return (0, utils_2.compile)([clientConfig, ...additionalPluginConfigs]);
         }
         else {
-            return (0, utils_2.compile)([clientConfig, serverConfig]);
+          return (0, utils_2.compile)([clientConfig, serverConfig, ...additionalPluginConfigs]);
         }
     });
     const { collectedData } = await utils_3.PerfLogger.async('SSG', () => executeSSG({
@@ -210,24 +210,24 @@ async function getBuildClientConfig({ props, cliOptions, }) {
         minify: cliOptions.minify ?? true,
         bundleAnalyzer: cliOptions.bundleAnalyzer ?? false,
     });
-    let { config } = result;
-    config = (0, configure_1.executePluginsConfigureWebpack)({
+    let { config: tmpConfig } = result;
+    let [ config, additionalPluginConfigs ] = (0, configure_1.executePluginsConfigureWebpack)({
         plugins,
-        config,
+        config: tmpConfig,
         isServer: false,
         jsLoader: props.siteConfig.webpack?.jsLoader,
     });
-    return { clientConfig: config, clientManifestPath: result.clientManifestPath };
+    return { clientConfig: config, additionalPluginConfigs, clientManifestPath: result.clientManifestPath };
 }
 async function getBuildServerConfig({ props }) {
     const { plugins } = props;
     const result = await (0, server_1.default)({
         props,
     });
-    let { config } = result;
-    config = (0, configure_1.executePluginsConfigureWebpack)({
+    let { config: tmpConfig } = result;
+    let [ config ] = (0, configure_1.executePluginsConfigureWebpack)({
         plugins,
-        config,
+        config: tmpConfig,
         isServer: true,
         jsLoader: props.siteConfig.webpack?.jsLoader,
     });
diff --git a/node_modules/@docusaurus/core/lib/commands/start/webpack.js b/node_modules/@docusaurus/core/lib/commands/start/webpack.js
index 1105ced..7fc6a8e 100644
--- a/node_modules/@docusaurus/core/lib/commands/start/webpack.js
+++ b/node_modules/@docusaurus/core/lib/commands/start/webpack.js
@@ -99,26 +99,26 @@ async function createDevServerConfig({ cliOptions, props, host, port, }) {
 }
 async function getStartClientConfig({ props, minify, poll, }) {
     const { plugins, siteConfig } = props;
-    let { clientConfig: config } = await (0, client_1.createStartClientConfig)({
+    let { clientConfig: tmpConfig } = await (0, client_1.createStartClientConfig)({
         props,
         minify,
         poll,
     });
-    config = (0, configure_1.executePluginsConfigureWebpack)({
+    let [config, additionalPluginConfigs] = (0, configure_1.executePluginsConfigureWebpack)({
         plugins,
-        config,
+        config: tmpConfig,
         isServer: false,
         jsLoader: siteConfig.webpack?.jsLoader,
     });
-    return config;
+    return [config, additionalPluginConfigs];
 }
 async function createWebpackDevServer({ props, cliOptions, openUrlContext, }) {
-    const config = await getStartClientConfig({
+  const [config, additionalPluginConfigs] = await getStartClientConfig({
         props,
         minify: cliOptions.minify ?? true,
         poll: cliOptions.poll,
     });
-    const compiler = (0, webpack_1.default)(config);
+    const compiler = (0, webpack_1.default)([config, ...additionalPluginConfigs]);
     registerWebpackE2ETestHook(compiler);
     const defaultDevServerConfig = await createDevServerConfig({
         cliOptions,
diff --git a/node_modules/@docusaurus/core/lib/webpack/configure.js b/node_modules/@docusaurus/core/lib/webpack/configure.js
index 9cc1639..cdbfeac 100644
--- a/node_modules/@docusaurus/core/lib/webpack/configure.js
+++ b/node_modules/@docusaurus/core/lib/webpack/configure.js
@@ -77,12 +77,18 @@ function executePluginsConfigurePostCss({ plugins, config, }) {
 function executePluginsConfigureWebpack({ plugins, config, isServer, jsLoader, }) {
     // Step1 - Configure Webpack
     let resultConfig = config;
+    const additionalPluginConfigs = [];
     plugins.forEach((plugin) => {
-        const { configureWebpack } = plugin;
+      const { configureWebpack, configureAdditionalWebpack } = plugin;
         if (configureWebpack) {
             resultConfig = applyConfigureWebpack(configureWebpack.bind(plugin), // The plugin lifecycle may reference `this`.
             resultConfig, isServer, jsLoader, plugin.content);
         }
+        if (configureAdditionalWebpack) {
+          const newConfig = applyConfigureWebpack(configureAdditionalWebpack.bind(plugin),
+            {}, false, jsLoader, plugin.content);
+          additionalPluginConfigs.push(newConfig);
+        }
     });
     // Step2 - For client code, configure PostCSS
     // The order matters! We want to configure PostCSS on loaders
@@ -95,5 +101,5 @@ function executePluginsConfigureWebpack({ plugins, config, isServer, jsLoader, }
             config: resultConfig,
         });
     }
-    return resultConfig;
+    return [resultConfig, additionalPluginConfigs];
 }
